src/main/java/ch/uzh/ifi/hase/soprafs22/game/logger/GameStatistics.java

```diff
@@ -0,0 +1,29 @@
+package ch.uzh.ifi.hase.soprafs22.game.logger;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Provides statistical information to the client about a given game.
+ */
+public interface GameStatistics {
+    /**
+     * Returns the number of life units per player at each turn.
+     */
+    Map<Long, List<Integer>> unitsPerPlayer();
```

Comment:
a Map of a Long and a List<Integer>?

Don't we have something from the PlayerDecorator already at hand to count the units of a player?

Reply:
Yes and no. This is exactly the information we need to keep track of the stats. PlayerDecorator has much more information (such as the units themselves, and taking a snapshot of that has a much higher memory footprint). 
Moreover, keeping the counter separate, allows for separation of responsabilities (see https://java-design-patterns.com/principles/#single-responsibility-principle).